client/src/pages/Details/index.tsx

```diff
@@ -21,15 +14,13 @@ const Details: React.FC = () => {
   useEffect(() => {
     const getLocationData = async () => {
       if (location) {
-        const locationName = extractLocationName(location);
-        const searchLocationData = await getLocationByQuery(locationName);
-        if (searchLocationData) {
-          const searchLocationCoords = {
-            lat: searchLocationData?.locations[0].lat,
-            lon: searchLocationData?.locations[0].lon
-          };
-          setCoordinates(searchLocationCoords);
-        }
+        const lon = location.split('(')[1].split(')')[0];
+        const lat = location.split('(')[2].split(')')[0];
```

Comment:
please create shared function for parsing saved search string and reuse it everywhere